### PR TITLE
ReactionLib refactor

### PR DESCRIPTION
**Pull request by RedlineTriad**
### Purpose
Refactors ReactionLib.cs, splits up the monotlithic function into multiple smaller ones<br>

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### CODE REVIEW

File: UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs [hunk 1]
```diff
@@ -18,175 +16,233 @@
...
+					float.Parse(reaction["Minimum_temperature"].ToString()) :
+					0.0f
+			});
 		}
-		Logger.Log ("JsonImportInitialization done!", Category.Chemistry);
+		Logger.Log("JsonImportInitialization done!", Category.Chemistry);
 	}
 
-	private static void CemInitialization ()
+	private static void CemInitialization()
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs]: Old typo?
```suggestion
	private static void ChemInitialization()
```

File: UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs [hunk 2]
```diff
@@ -18,175 +16,233 @@
...
+			{
+				return false;
+			}
 		}
-		return (Area);
+		return true;
 	}
 
-	private static float SwapFix (float n1, float n2)
+	private static float SwapFix(float n1, float n2)
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs:228]: Would be useful to explain what this is for and why it's needed (if you know)

**Review by Jibbajabbafic — CHANGES_REQUESTED** [review]
Good stuff.
Thanks for refactoring this!

I only have some minor changes which I think should be added

I also tested the ammonia reaction with 5u of hydrogen and 5u nitrogen and it made 5u of ammonia before this PR so it was already broken.

File: UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs (hunk 2 above)
Comment (RedlineTriad) [reply UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs:228]: You would have to ask @Bod9001 I think.

Comment (Jibbajabbafic) [reply UnityProject/Assets/Scripts/Chemistry/ReactionLib.cs:228]: It'll do without

**Review by Jibbajabbafic — APPROVED** [review]

**Pull request merged**